### PR TITLE
Remove hardcoded protocol for notification page URL

### LINKED ISSUE #198: https should not be hard-coded in the notification page URL
I am using Firefox extension with enterprise github and it is not https-enabled (I know this is bad, but this is what it is). Pretty much everything works, except the opening notifications page when clicking on add-in icon - there is `https` hardcoded in the URL construction: https://github.com/sindresorhus/notifier-for-github/blob/master/source/lib/api.js#L17. Can we somehow use protocol from the `${rootUrl}` provided in add-in settings? Thanks!

### PR DESCRIPTION
**Pull request by Hydrog3n**
Fixes #198 

I created another function who gets the `rootUrl` and processing like removing the last `/` if exist and if `api.github.com` it's will return github website.

Maybe the better way is to create another option?

### CODE REVIEW

File: source/lib/api.js [hunk 1]
```diff
@@ -28,6 +28,16 @@ export async function getApiUrl() {
 	return `${rootUrl}api/v3`;
 }
 
+export async function getRootUrl() {
+	const {rootUrl} = await optionsStorage.getAll();
+
+	if (/(^(https:\/\/)?(api\.)?github\.com)/.test(rootUrl)) {
```
Comment (notlmn) [inline source/lib/api.js]: I think it's a good time for us to move from regex check here to proper URL parsing

Something like this:

```suggestion
	if ((new URL(rootUrl)).origin === 'https://api.github.com) {
```

**Review by notlmn — CHANGES_REQUESTED** [review]
I like this change, non-destructive for most and we already ask user to include protocol as well in input.

**Comment by fregante** [comment]
You should replace `getHostname` instead of adding a new function; it does the same thing and is used everywhere already.

It should probably be renamed for clarity too

```js
export async function getGitHubOrigin() {
	const {rootUrl} = await optionsStorage.getAll();
	const {origin} = new URL(rootUrl);
	return origin === 'https://api.github.com' ? 'https://github.com' : origin;
}
```

**Comment by Hydrog3n** [comment]
Thank you @notlmn I don't have time to continue my PR sorry.

**Comment by notlmn** [comment]
No problem @Hydrog3n, I am just cleaning up stuff that have been stale for too long. Thanks for you other contributions.

**Comment by notlmn** [comment]
We should actually use `https://github.com` in options for `rootUrl` instead of `https://api.github.com`. Ideally should be done through migrations.

Merging this for now.

**Pull request merged**